sdk/containerregistry/azure-containerregistry/azure/containerregistry/_download_stream.py

```diff
@@ -0,0 +1,82 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import hashlib
+from typing import Iterator, ContextManager, TypeVar, cast, Tuple, Dict, Any
+from typing_extensions import Protocol
+from azure.core.pipeline import PipelineResponse
+
+T = TypeVar('T', bound='DownloadBlobStream')
+
+
+class GetNext(Protocol):
+    def __call__(self, *args: Any, range_header: str) -> Iterator[bytes]:
+        pass
+
+
+class DownloadBlobStream(
+    Iterator[bytes],
+    ContextManager[T],
+):
+    """Protocol for methods to provide streamed responses."""
+
+    def __init__(
+        self,
+        *,
+        response: PipelineResponse,
+        get_next: GetNext,
+        blob_size: int,
+        downloaded: int,
+        digest: str,
+        chunk_size: int
+    ) -> None:
+        self._response = response
+        self._response_bytes = response.http_response.iter_bytes()
+        self._next = get_next
+        self._blob_size = blob_size
+        self._downloaded = downloaded
+        self._digest = digest
+        self._chunk_size = chunk_size
+        self._hasher = hashlib.sha256()
+
+    def __enter__(self: T) -> T:
```

Comment:
You can import `Self` from `typing_extensions` for this.
```suggestion
    def __enter__(self) -> Self:
```

Reply:
Updated both two comments above.